checks/__init__.py

```diff
@@ -502,6 +512,18 @@ def get_service_checks(self):
         self.service_checks = []
         return service_checks
 
+    def get_service_metadata(self):
+        """
+        Return a list of the metadata dictionaries saved by the check -if any-
+        and clears them out of the instance's service_checks list
+
+        @return the list of metadata saved by this check
+        @rtype list of metadata dicts
+        """
+        service_metadata = self.service_metadata
+        self.service_metadata = []
```

Comment:
i think service_metadata should be a
```python
defaultdict(list)
```

use case: You have multiple versions of your application running on your instance